make build fails with Docker due to --volume flag

### Issue Description

It looks like the following commit introduced a --volume flag that breaks Docker-based builds:
https://github.com/containers/ramalama/commit/6c59fd7fd50ffe0048f1558f2062f0bbe76f706e

As a result, `make build` no longer works correctly in Docker environments.

### Steps to reproduce the issue

`make build` with Docker installed

### Describe the results you received

docker build --no-cache --platform linux/x86_64 --volume=/home/xiaodongye/ws/ggml/ramalama:/run/ramalama --security-opt=label=disable -t quay.io/ramalama/ramalama -f ramalama/Containerfile .
unknown flag: --volume
See 'docker buildx build --help'.
make: *** [Makefile:104: build] Error 125

### Describe the results you expected

`make build` pass

### ramalama info output

```yaml
N/A
```

### Upstream Latest Release

Yes

### Additional environment details

_No response_

### Additional information

_No response_

--- a/container-images/scripts/build-cli.sh
+++ b/container-images/scripts/build-cli.sh
@@ -24,7 +24,9 @@ dnf_install() {
 install_ramalama() {
   # link podman-remote to podman for use by RamaLama
   ln -sf /usr/bin/podman-remote /usr/bin/podman
-  python3 -m pip install /run/ramalama --prefix=/usr
+  if [ -e "/run/ramalama" ]; then
+    python3 -m pip install /run/ramalama --prefix=/usr
+  fi
 }
 
 main() {

--- a/container_build.sh
+++ b/container_build.sh
@@ -29,8 +29,14 @@ add_build_platform() {
 
 
   conman_build+=("--platform" "$platform")
-  conman_build+=("--volume=${RAMALAMA_DIR}:/run/ramalama")
-  conman_build+=("--security-opt=label=disable")
+  case $conman_bin in
+    podman)
+      conman_build+=("--volume=${RAMALAMA_DIR}:/run/ramalama")
+      conman_build+=("--security-opt=label=disable")
+      ;;
+    docker)
+      ;;
+  esac
   if [ -n "$version" ]; then
       conman_build+=("--build-arg" "VERSION=$version")
       conman_build+=("-t" "$REGISTRY_PATH/${target}-${version}")

